sstables/kl/reader.cc

```diff
@@ -33,7 +33,12 @@ class mp_row_consumer_reader_k_l : public mp_row_consumer_reader_base, public fl
         : mp_row_consumer_reader_base(std::move(sst))
         , impl(std::move(s), std::move(permit))
         , _rtc_gen(*_schema)
-    {}
+    {
+        _permit.on_start_sstable_read();
+    }
+    virtual ~mp_row_consumer_reader_k_l() {
+        _permit.on_finish_sstable_read();
+    }
```

Comment:
Move to mp_row_consumer_reader_base?

Reply:
That doesn't have access to a permit.

Reply:
perhaps we can later move the permit into it? but that doesn't have to be done now, of course.

Reply:
Not worth the churn.